Classes/Domain/Model/Changes/AbstractMove.php

```diff
@@ -67,4 +67,20 @@ protected function generateUniqueNodeName(NodeInterface $parentNode)
         return $this->contentRepositoryNodeService
             ->generateUniqueNodeName($parentNode->getPath());
     }
+
+    /**
+     * Generate a unique node name for the copied node if needed
+     *
+     * @param NodeInterface $parentNode
+     * @param NodeInterface $node
+     * @return string|null
```

Comment:
I'm always very skeptical when it comes to mixed return types. It probably makes sense here, but we should document what `null` means exactly

Reply:
..and what "if needed" means in this case

Reply:
```suggestion
     * @return string|null If the current name can be used, null is returned
```